libsolidity/ast/Types.cpp

```diff
@@ -1992,20 +2004,33 @@ bool StructType::operator==(Type const& _other) const
 	return ReferenceType::operator==(other) && other.m_struct == m_struct;
 }
 
+
 unsigned StructType::calldataEncodedSize(bool) const
 {
+	solAssert(!isDynamicallyEncoded(), "");
+
 	unsigned size = 0;
 	for (auto const& member: members(nullptr))
 		if (!member.type->canLiveOutsideStorage())
 			return 0;
```

Comment:
Can you try adding a failing assertion here?

Reply:
Weird - I thought I did that already earlier... it should definitely assert here, yes.

Reply:
Ah, no, apparently I only did for the tail size - I'll change it here as well, it should work.